examples/nlp/mlm_and_finetune_with_bert.py

```diff
@@ -0,0 +1,290 @@
+        value = tf.keras.layers.Dense(flags.EMBED_DIM)(embedding_layer)
+        key = tf.keras.layers.Dense(flags.EMBED_DIM)(embedding_layer)
+
+        # multi headed self-attention
+        attention_output = tf.keras.layers.MultiHeadAttention(
+            num_heads=flags.NUM_HEAD, key_dim=flags.EMBED_DIM // flags.NUM_HEAD
+        )(query, value, key)
+        attention_output = tf.keras.layers.Dropout(0.1)(attention_output)
+        attention_output = tf.keras.layers.LayerNormalization(epsilon=1e-6)(
+            embedding_layer + attention_output
+        )
+
+        # feed-forward leyer
+        ffn = tf.keras.Sequential(
+            [
+                tf.keras.layers.Dense(flags.FF_DIM, activation="relu"),
+                tf.keras.layers.Dense(flags.EMBED_DIM),
+            ]
+        )
+        ffn_output = ffn(attention_output)
+        ffn_output = tf.keras.layers.Dropout(0.1)(ffn_output)
+        sequence_output = tf.keras.layers.LayerNormalization(epsilon=1e-6)(
+            attention_output + ffn_output
+        )
+
+        pooled_output = tf.keras.layers.GlobalMaxPooling1D()(ffn_output)
+
+        return sequence_output, pooled_output
+
+
+def build_mlm_model():
+    tf.keras.backend.clear_session()
+    inputs = tf.keras.layers.Input((flags.MAX_LEN,), dtype=tf.int32)
+    sequence_output, pooled_output = BERT(flags)(inputs)
+    outputs = tf.keras.layers.Dense(flags.VOCAB_SIZE)(sequence_output)
+    mlm_model = tf.keras.Model(inputs, outputs)
+
+    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(
+        reduction=tf.keras.losses.Reduction.NONE, from_logits=True
+    )
+    optimizer = tf.keras.optimizers.Adam(learning_rate=flags.LR)
+    mlm_model.compile(optimizer=optimizer, loss=loss_fn)
+    return mlm_model
+
+
+def build_classifer_model():
+    tf.keras.backend.clear_session()
+    inputs = tf.keras.layers.Input((flags.MAX_LEN,), dtype=tf.int32)
+    sequence_output, pooled_output = BERT(flags)(inputs)
+    outputs = tf.keras.layers.Dense(1, activation="sigmoid")(pooled_output)
+    classifer_model = tf.keras.Model(inputs, outputs)
+    optimizer = tf.keras.optimizers.Adam(learning_rate=flags.LR)
+    classifer_model.compile(
+        optimizer=optimizer, loss="binary_crossentropy", metrics=["accuracy"]
+    )
+    return classifer_model
+
+
+def get_end_to_end(model):
+    inputs_string = tf.keras.Input(shape=(1,), dtype="string")
+    indices = vectorize_layer(inputs_string)
+    outputs = model(indices)
+    end_to_end_model = tf.keras.Model(inputs_string, outputs, name="end_to_end")
+    return end_to_end_model
+
+
+"""
+## Build, Train, Save MLM
+"""
+
+mlm_model = build_mlm_model()
+mlm_model.fit(mlm_train_data, epochs=1)
+mlm_model.save("mlm_model.h5")
+
+"""
+## Build, Train, Save Fine-Tune Model
+"""
+
+classifer_model = build_classifer_model()
+classifer_model.load_weights("mlm_model.h5", skip_mismatch=True, by_name=True)
+classifer_model.fit(classifier_train_data, epochs=1)
```

Comment:
Please add a section about model evaluation and using the model in practice on a new sample.

Reply:
ok